python/sglang/srt/layers/attention/flashinfer_backend.py

```diff
@@ -430,6 +437,37 @@ def forward_extend(
                 logits_soft_cap=logits_soft_cap,
             )
 
+            qo_indptr = self.prefill_wrapper_ragged._qo_indptr_buf
+            otest = torch.empty_like(o1)
+            max_extend_len = qo_indptr
+            max_extend_len = torch.max(max_extend_len[1:] - max_extend_len[:-1]).item()
+            extend_attention_fwd(
+                q.view(-1, layer.tp_q_head_num, layer.qk_head_dim),
+                k.contiguous(),
+                v.contiguous(),
+                otest.view(-1, layer.tp_q_head_num, layer.v_head_dim),
+                forward_batch.token_to_kv_pool.get_key_buffer(layer.layer_id),
+                forward_batch.token_to_kv_pool.get_value_buffer(layer.layer_id),
+                qo_indptr,
+                torch.zeros_like(qo_indptr),
+                torch.tensor([], dtype=torch.int32, device=q.device),
+                None,
+                None,
+                max_extend_len,
+                layer.scaling,
+                layer.logit_cap,
+            )
+
+            cos_sim = torch.cosine_similarity(otest.view_as(q), o1.view_as(q), dim=-1)
+            print(f"layer {self.layer} min cos_sim = {cos_sim.min().item()}")
+
+            # error occurred in prefill_wrapper_ragged's o, but lse seems correct
+            if not torch.all(cos_sim >= torch.full_like(cos_sim, 0.999)):
```

Comment:
Here is some code for error detection XD.
Error occurred in prefill_wrapper_ragged's o at the first layer.